Library/Services/Implementation/NfieldThemesService.cs

```diff
@@ -43,20 +43,22 @@ public async Task UploadThemeAsync(string templateId, string themeName, string f
             var fileName = Path.GetFileName(filePath);
 
             if (!File.Exists(filePath))
-                throw new FileNotFoundException(fileName);
-
-            using (var byteArrayContent = new ByteArrayContent(File.ReadAllBytes(filePath)))
             {
-                byteArrayContent.Headers.ContentType = new MediaTypeHeaderValue("application/octet-stream");
-                var uri = GetUploadThemeUri(templateId, themeName);
-                _ = await Client.PutAsync(uri, byteArrayContent)
-                       .ContinueWith(responseMessageTask => responseMessageTask.Result.Content.ReadAsStringAsync().Result)
-                       .ContinueWith(stringResult => JsonConvert.DeserializeObject<BackgroundActivityStatus>(stringResult.Result).ActivityId)
-                       .ContinueWith(activityResult => ConnectionClient.GetActivityResultAsync<int>(activityResult.Result, "Status"))
-                       .Unwrap()
-                       .FlattenExceptions()
-                       .ConfigureAwait(false);
+                throw new FileNotFoundException(fileName);
             }
+
+            var themeContent = File.ReadAllBytes(filePath);
+
+            await DoUploadThemeAsync(templateId, themeName, themeContent);
+        }
+
+        public async Task UploadThemeAsync(string templateId, string themeName, byte[] themeContent)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(templateId, nameof(templateId));
+            Ensure.ArgumentNotNullOrEmptyString(themeName, nameof(themeName));
+            Ensure.ArgumentNotNull(themeContent, nameof(themeContent));
+
+            await DoUploadThemeAsync(templateId, themeName, themeContent).ConfigureAwait(false);
```

Comment:
I noticed that all over the SDK we are very inconsistent with using ConfigureAwait. It seems to work with and without it.
Is there a good reason to use it?

Reply:
I did it here because it was already done for this class.
But I think it is good practice in a library where you do not know where it will be consumed.
When you would for example call this from a windows forms application and would not use configureawait(false) everything would be executed on the UI thread.